packages/protocol/contracts/stability/Broker.sol

```diff
@@ -0,0 +1,164 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { console2 as console } from "forge-std/console2.sol";
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { IERC20 } from "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+import { IPairManager } from "./interfaces/IPairManager.sol";
+import { IBroker } from "./interfaces/IBroker.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+import { IMentoExchange } from "./interfaces/IMentoExchange.sol";
+
+import { StableToken } from "./StableToken.sol";
+
+import { Initializable } from "../common/Initializable.sol";
+import { IStableToken } from "./interfaces/IStableToken.sol";
+
+/**
+ * @title Broker
+ * @notice The broker executes swaps and keeps track of spending limits per pair
+ */
+contract Broker is IBroker, Initializable, Ownable {
+  /* ==================== State Variables ==================== */
+
+  // Address of the broker contract.
+  IPairManager public pairManager;
+
+  // Address of the reserve.
+  IReserve public reserve;
+
+  /* ==================== Constructor ==================== */
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialization.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Allows the contract to be upgradable via the proxy.
+   * @param _pairManager The address of the PairManager contract.
+   * @param _reserve The address of the Rezerve contract.
+   */
+  function initilize(address _pairManager, address _reserve) external initializer {
+    _transferOwnership(msg.sender);
+    setPairManager(_pairManager);
+    setReserve(_reserve);
+  }
+
+  /* ==================== View Functions ==================== */
+
+  function quote(bytes32 pairId, address tokenIn, uint256 amountIn)
+    external
+    view
+    returns (address tokenOut, uint256 amountOut)
+  {
+    (tokenOut, amountOut, , , ) = _quote(pairId, tokenIn, amountIn);
+  }
+
+  /* ==================== Mutative Functions ==================== */
+
+  /**
+   * @notice Update the PairManager address
+   * @dev only callable by owner
+   * @param _pairManager The new pair manager address
+   */
+  function setPairManager(address _pairManager) public onlyOwner {
+    require(_pairManager != address(0), "PairManager address must be set");
+    emit PairManagerUpdated(_pairManager, address(pairManager));
```

Comment:
Sure, for me it doesn't really make a difference when the emit occurs because it gets rolled back in case of a revert anyway. I put it there to avoid some gas costs when storing the previous address. But that's not a big deal, also I can probably remove the previous address from the event, I'm just used to emitting both on these updates for clarity but it's not super necessary.

I'm kinda fine with anything so which would you prefer:
(a) Keep the `prevAddress` in the event and use a temp variable in order to emit after the state change
(b) Remove the `prevAddress` from the vent and emit after the state change without needing a temp variable
(c) Keep as is

Reply:
Okay got it. May as well keep it as it is.